packages/hoppscotch-selfhost-web/src/platform/collections/collections.platform.ts

```diff
@@ -284,6 +284,20 @@ function setupUserCollectionCreatedSubscription() {
         return
       }
 
+      // While duplicating a collection, the new entry added to the store has an ID with a suffix to be updated after the backend ID is received from the GQL subscription
+      // This is to prevent the new entry from being added to the store again when the GQL subscription
+      // The boolean return value indicates if the GQL subscription was fired because of a duplicate collection action and whether the collection should be added to the store
+      const shouldCreateCollection = issueBackendIDToDuplicatedCollection(
+        collectionStore,
+        collectionType,
+        userCollectionBackendID,
+        parentCollectionID
+      )
+
+      if (!shouldCreateCollection) {
+        return
+      }
```

Comment:
For creation actions except duplication the new collection entry received from the GQL subscription will be added to the store as indicated by the return value from the `issueBackendIDToDuplicatedCollection()` helper function.